aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+
+.. literalinclude::   forms/sample/final/Views/Home/Index.cshtml
+  :language: HTML
+  :lines: 4
+  :dedent: 3
+  
+Consider the following view model:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryViewModel.cs
+  :language: c#
+  
+The ``Index`` method initializes the ``CountryViewModel``, sets the selected country and passes it to the ``Index`` view.
+
+.. literalinclude:: forms/sample/final/Controllers/HomeController.cs
+  :language: c#
+  :lines: 8-13
+  :dedent: 6
+
+The HTTP POST ``Index`` method displays the selection:
+
+.. literalinclude:: forms/sample/final/Controllers/HomeController.cs
+  :language: c#
+  :lines: 15-27
+  :dedent: 6
+  
+The ``Index`` view:
+
+.. literalinclude::   forms/sample/final/Views/Home/Index.cshtml
+  :language: HTML
+  :emphasize-lines: 4
+  
+Which generates the following HTML:
+
+.. code-block:: HTML  
+  :linenos:
+  :emphasize-lines: 2-6 
+
+  <form method="post" action="/">
+    <select id="Country" name="Country">
+      <option value="MX">Mexico</option>
+      <option selected="selected" value="CA">Canada</option>
+      <option value="US">USA</option>
+    </select> 
+      <br /><button type="submit">Register</button>
+    <input name="__RequestVerificationToken" type="hidden" value="<removed for brevity>" />
+  </form>
+
+:Note: We do not recommend using ``ViewBag`` or ``ViewData`` with the Select Tag Helper. A view model is more robust at providing MVC metadata and generally less problematic. 
+
+The ``asp-for`` attribute value doesn't require a ``Model`` prefix, while ``asp-items does``:
+
+.. literalinclude::   forms/sample/final/Views/Home/Index.cshtml
+  :language: HTML
+  :lines: 4
+  :dedent: 3
+
+The ``asp-for`` attribute value is the right hand side of a lambda expression. For example, ``asp-for=""Property1"`` becomes ``m => m.Property1`` in the generated code which is why you don't need to prefix with ``Model``.
+  
+Enum binding
+^^^^^^^^^^^^^^
+
+It's often convenient to use an ``enum`` for a list. The ``CountryEnumViewModel`` uses the ``CountryEnum`` to generate a list of countries:
+
+.. literalinclude::  forms/sample/final/ViewModels/CountryEnumViewModel.cs
+  :language: c#
+  :lines: 3-6
+  :dedent: 3
+  
+.. literalinclude::  forms/sample/final/ViewModels/CountryEnum.cs
+  :language: c#
+  :lines: 1-4,6,8-
+
+The `GetEnumSelectList <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/IHtmlHelper/index.html>`__ method generates a `SelectList <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/SelectList/index.html>`__ object for an enum.
+
+.. literalinclude::   forms/sample/final/Views/Home/IndexEnum.cshtml
+  :language: HTML
+  :emphasize-lines: 4-6
```

Comment:
Unclear why the `<select>` element is highlighted when this sub-section is primarily about `Html.GetEnumSelectList<TEnum>()`.